### PR TITLE
Rework button documentation, changes to buttons, and new way to display components

### PR DESCRIPTION
**Pull request by pthoang**
## Description
Fixes [DG-551](https://payexjira.atlassian.net/secure/RapidBoard.jspa?rapidView=80&projectKey=DG&modal=detail&selectedIssue=DG-551)
Fixes [DG-516](https://payexjira.atlassian.net/secure/RapidBoard.jspa?rapidView=80&projectKey=DG&modal=detail&selectedIssue=DG-516)
Fixes [DG-507](https://payexjira.atlassian.net/secure/RapidBoard.jspa?rapidView=80&projectKey=DG&modal=detail&selectedIssue=DG-507)

Also added a showCasePanelAdvanced component that will be used with componentPreview

## How Has This Been Tested?
Ran all existing tests, added testing for showCasePanelAdvanced

## Screenshots (if appropriate):

image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have updated the **CHANGELOG** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/App/components/Button/index.js  4
         

Clones added
============
- src/App/docutils/ComponentPreview/index.test.js  2
- src/App/docutils/ComponentPreview/index.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/design.swedbankpay.com/pullRequest?prid=6148238&bid=20093938)

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/App/components/Button/index.js  4
         

Clones added
============
- src/App/docutils/ComponentPreview/index.test.js  2
- src/App/docutils/ComponentPreview/index.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/design.swedbankpay.com/pullRequest?prid=6148238&bid=20093938)

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/App/components/Button/index.js  4
         

Clones added
============
- src/App/docutils/ComponentPreview/index.test.js  2
- src/App/docutils/ComponentPreview/index.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/design.swedbankpay.com/pullRequest?prid=6148238&bid=20093938)

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/App/components/Button/index.js  4
         

Clones added
============
- src/App/docutils/ComponentPreview/index.test.js  2
- src/App/docutils/ComponentPreview/index.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/design.swedbankpay.com/pullRequest?prid=6148238&bid=20093938)

File: src/App/docutils/ComponentPreview/index.js [hunk 1]
```diff
@@ -153,10 +153,163 @@ const ComponentPreview = ({ children, language, removeOuterTag, hideValue, remov
         </div>
     );
 
+    const { tabs } = window.dg;
+
+    class ShowCasePanelAdvanced extends Component {
```
Comment (VikingTristan) [inline src/App/docutils/ComponentPreview/index.js:158]: Complicated, but effective class. As you said on slack, we can later go over how to make the developer experience, especially regarding props, as easy as possible.

File: src/less/documentation-swedbankpay.less [hunk 2]
```diff
@@ -498,6 +498,123 @@
...
+                    .options-close {
+                        cursor: pointer;
+                    }
+                }
+
+                .options-body {
+                    padding: 1rem;
+                }
+            }
+        }
```
Comment (VikingTristan) [inline src/less/documentation-swedbankpay.less:617]: Perhaps this class can be in a shared documentation file as we discussed on slack.

**Review by VikingTristan — APPROVED** [review]
Looking good!

File: src/less/documentation-swedbankpay.less (hunk 2 above)
Comment (VikingTristan) [reply src/less/documentation-swedbankpay.less:617]: "This class" here refers to `&.showcase-panel-advanced`

**Pull request merged**